jib-core/src/main/java/com/google/cloud/tools/jib/image/json/BuildableManifestTemplate.java

```diff
@@ -61,7 +61,6 @@
     @SuppressWarnings("unused")
     private ContentDescriptorTemplate() {}
 
-    @VisibleForTesting
```

Comment:
For this particular one, maybe SonarCloud is confused. Making this package-private doesn't break any non-test code. It's `public` only because some tests need to access it. Maybe we should file a bug against SonarCloud?

However, I am fine with just removing the annotation if it's more of an unnecessary chore to suppress this false positive. It's a JSON template class, and it's totally OK to have getters as public. We just didn't make it `public` at the beginning based on the principle of the least privilege.

Reply:
I see. As you mentioned, the annotation is only used for the purpose of the class member being accessible to tests. Since this is a public getter, I think it is still worth removing the annotation as it is not needed for that purpose anymore?

Reply:
> the annotation is only used for the purpose of the class member being accessible to tests. ... as it is not needed for that purpose

The annotation is used for that purpose here. If there were no tests, it could be package-private instead of `public`. The only reason we make it `public` is that there is a test class wanting to use it.

However, as I said, I am fine with assuming that conceptually this class can naturally have public getters even though no prod code currently needs `public` access.

Reply:
> Having both `public` and `@VisibleForTesting` seems a little redundant, unless I'm missing something here.

This is what is always done when you expose a non-public method to a test that needs it. That is, adding `public` and `@VisibleForTesting` for a non-public method.

Reply:
Oh that makes a lot of sense now. In that case, I think you are right that this particular one should be marked as a false positive.

Reply:
I'm actually OK with removing it. Your call.

Reply:
Ah okay. I had a hard time making a decision on it because I've seen some places where we specify the productionVisibility for the code being annotated (like https://developer.android.com/reference/androidx/annotation/VisibleForTesting) which seems like a good practice. However, since that option is not available for guava I'll just remove it for now to silence sonar and avoid confusion in the future.

Reply:
BTW, it is unfortunate that SonarCloud cannot catch this. Sort of a bug there.

Reply:
You're right, this is indeed a limitation of the tool. I added a comment on the sonarcloud cloud issue explaining this but I'll also look into see if I can file a bug with them.